### PR TITLE
Split cpu and reference implementation

### PR DESCRIPTION
**Pull request by pfultz2**
It doesn't enable the cpu target yet(I plan to do that in a future PR).

* Converts `cpu::target` to `ref::target`
* Update driver so it can build and verify multiple targets
* Update gpu_ops test to run for multiple targets
* Split the verify tests into seperate cpp files

### CODE REVIEW

File: src/targets/ref/lowering.cpp [hunk 1]
```diff
@@ -0,0 +1,974 @@
...
+                    });
+                });
+        }
+
+        return output;
+    }
+};
+MIGRAPHX_REGISTER_OP(cpu_batch_norm_inference)
+
+struct cpu_lrn
```
Comment (scxiao) [inline src/targets/ref/lowering.cpp:130]: should we rename the name to ref_lrn here?

File: test/quantization.cpp [hunk 2]
```diff
@@ -1009,7 +1009,7 @@ TEST_CASE(int8_quantization_dot)
         m["a"] = migraphx::generate_argument(sa);
         m["c"] = migraphx::generate_argument(sc);
         std::vector<float> quant_result;
-        migraphx::target cpu_t = migraphx::cpu::target{};
+        migraphx::target cpu_t = migraphx::ref::target{};
```
Comment (scxiao) [inline test/quantization.cpp:1012]: Should we also cpu_t to ref_t here?

**Comment by pfultz2** [comment]
I dont think its necessary to add tests fro env functions, so I think this can be merged in as well.

**Review by kahmed10 — APPROVED** [review]

File: test/gpu/quantization.cpp [hunk 3]
```diff
@@ -4,21 +4,21 @@
...
 #include <migraphx/onnx.hpp>
-#include "test.hpp"
+#include <test.hpp>
 #include <migraphx/half.hpp>
 
 TEST_CASE(gpu_target_copy)
 {
     migraphx::target gpu_t = migraphx::gpu::target{};
-    migraphx::target cpu_t = migraphx::cpu::target{};
+    migraphx::target cpu_t = migraphx::ref::target{};
```
Comment (scxiao) [inline test/gpu/quantization.cpp:21]: maybe change the cpu_t to ref_t here?

File: test/gpu/quantization.cpp [hunk 4]
```diff
@@ -4,21 +4,21 @@
...
     migraphx::shape s{migraphx::shape::int8_type, {2, 3, 4, 5}};
 
     auto cpu_arg_orig  = migraphx::generate_argument(s, 0x123456L);
```
Comment (scxiao) [inline test/gpu/quantization.cpp:24]: change to ref_arg_orig

File: test/gpu/quantization.cpp [hunk 5]
```diff
@@ -83,7 +83,7 @@ TEST_CASE(int8_quantization)
         m["b"] = migraphx::generate_argument(sb);
         m["c"] = migraphx::generate_argument(sc);
         std::vector<float> cpu_result;
```
Comment (scxiao) [inline test/gpu/quantization.cpp:85]: change to ref_result

File: test/gpu/quantization.cpp [hunk 6]
```diff
@@ -83,7 +83,7 @@ TEST_CASE(int8_quantization)
...
-        migraphx::target cpu_t = migraphx::cpu::target{};
+        migraphx::target cpu_t = migraphx::ref::target{};
```
Comment (scxiao) [inline test/gpu/quantization.cpp:86]: change to ref_t

File: test/gpu/quantization.cpp (hunk 3 above)
Comment (scxiao) [reply test/gpu/quantization.cpp]: I just changed that and the following

**Review by scxiao — APPROVED** [review]

**Comment by mvermeulen** [comment]
Is this one still pending?  Hard to tell if it passed all required steps.

**Comment by pfultz2** [comment]
It looks like the job got cancelled, I am rebuilding it.

**Pull request merged**